.github/scripts/read-binary-content.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash -e
+
+# This script creates a binary resource and verifies that its binary content
+# can be read.
+
+SCRIPT_DIR="$(dirname "$(readlink -f "$0")")"
+. "$SCRIPT_DIR/util.sh"
+
+BASE="http://localhost:8080/fhir"
+
+# 10 KiB of random data, base64 encoded
+DATA="$(openssl rand -base64 10240 | tr -d '\n')"
+
+echo "Ramdom Data: $DATA"
+
+binary() {
+cat <<END
+{
+  "resourceType": "Binary",
+  "contentType": "application/pdf",
+  "data": "$DATA"
+}
+END
+}
+
+# Create a Binary resource that contains that data, and get its ID
+ID=$(curl -s -H 'Content-Type: application/fhir+json' -d "$(binary)" "$BASE/Binary" | jq -r .id)
+
+echo "Created Binary resource that contains the Random Data, with ID: $ID"
+
+# Retrieve the Binary resource, and Base64 encode it so it can be safely handled by Bash
+BASE64_ENCODED_BINARY_RESOURCE=$(curl -s -H 'Accept: application/pdf' "$BASE/Binary/$ID" | base64 | tr -d '\n')
+
+echo "Binary Result, Base 64 encoded: $BASE64_ENCODED_BINARY_RESOURCE"
+
+echo "Binary data retrieved. Verifying content..."
+
+test "Base64 encoding of Resource data" "$DATA" "$BASE64_ENCODED_BINARY_RESOURCE"
```

Comment:
Please add a test reading the Binary resource using application/fhir+json and test that the base64 content is the same as the original content.